### ISSUE TITLE
Regression: DOM helpers no longer wait on test waiters unless `ember-test-helpers/wait` is imported

### ISSUE THREAD

**Issue by kybishop** [emberjs/ember-test-helpers#401]
I have been attempting to upgrade ember-attacher to ember 3.2 and beyond, but my tests started failing with the new test framework.

Previously, calls on `await click('.stuff')` would wait until all registered test waiters were settled. It appears this is no longer the case. Additionally, merely importing, not using, `ember-test-helpers/wait` (`import wait from 'ember-test-helpers/wait';`) causes tests in that file to pass again. From my sleuthing, importing wait registers the test waiters appropriately.

Am I right to think this is a regression, or was it unintentional that DOM helpers such as `click()` were waiting on the test waiters? I'm really banking on the former as I'd otherwise have to sprinkle my codebase with a ton of `await wait()`.

Thanks!

**Comment by rwjblue** [comment]
I’m generally not aware of a general issue like this (and haven’t seen/heard of others with the same symptoms), thanks for reporting!

How can we reproduce? Without some sort of reproduction, I believe this will be somewhat difficult to diagnose...

**Comment by rwjblue** [comment]
Can y’all share the output of:

```
npm ls @ember/test-helpers
npm ls ember-qunit
npm ls ember-cli-qunit 
```

**Comment by Frozenfire92** [comment]
```
npm ls @ember/test-helpers
...path...
└─┬ ember-cli-qunit@4.3.2
  └─┬ ember-qunit@3.3.2
    └── @ember/test-helpers@0.7.20
```

```
npm ls ember-qunit
...path...
└─┬ ember-cli-qunit@4.3.2
  └── ember-qunit@3.3.2
```

```
npm ls ember-cli-qunit
...path...
└── ember-cli-qunit@4.3.2
```

**Comment by rwjblue** [comment]
`ember-source@3.3` requires at least `@ember/test-helpers@0.7.22` (but you should just update to latest).

Specifically, you need https://github.com/emberjs/ember-test-helpers/pull/366.

**Comment by Frozenfire92** [comment]
Thanks for the help @rwjblue! definitely helped me come to a resolution

1) I don't think ember-source depends on the test helpers. Looking at [package.json](https://github.com/emberjs/ember.js/blob/master/package.json) so ensuring updating that didn't solve
2) I just ran `yarn` or `yarn install` which I found out won't update packages to newer resolutions

Looking in the yarn lock I was quite confused why it never went to the latest verison

```
ember-qunit@^3.3.2:
  version "3.3.2"
  resolved "https://registry.yarnpkg.com/ember-qunit/-/ember-qunit-3.3.2.tgz#cb48e9deaffa3b4c90983f28c5cf8590894c8ea3"
  dependencies:
    "@ember/test-helpers" "^0.7.18"
    broccoli-funnel "^2.0.1"
    broccoli-merge-trees "^2.0.0"
    common-tags "^1.4.0"
    ember-cli-babel "^6.3.0"
    ember-cli-test-loader "^2.2.0"
    qunit "^2.5.0"
```

```
"@ember/test-helpers@^0.7.18":
  version "0.7.20"
  resolved "https://registry.yarnpkg.com/@ember/test-helpers/-/test-helpers-0.7.20.tgz#488b1c8ef23626f8831e5cb750ffca86e017e6dc"
  dependencies:
    broccoli-funnel "^2.0.1"
    ember-cli-babel "^6.10.0"
    ember-cli-htmlbars-inline-precompile "^1.0.0"
```

Turns out my knowledge of yarn is a little lacking. The fix I did was to remove my yarn lock file and do a fresh `yarn install` but after googling I think `yarn upgrade` would have done it too.

Now in my lock file I have the latest test helpers and tests are passing (without my terrible hacks :smile: )
```
"@ember/test-helpers@^0.7.18":
  version "0.7.25"
  resolved "https://registry.yarnpkg.com/@ember/test-helpers/-/test-helpers-0.7.25.tgz#b4014c108b40ffaf74f3c4d5918800917541541d"
  dependencies:
    broccoli-funnel "^2.0.1"
    ember-cli-babel "^6.12.0"
    ember-cli-htmlbars-inline-precompile "^1.0.0"
```

**Comment by rwjblue** [comment]
Yep, `yarn` doesn't really expose a way to force transient dependency updates other than `yarn upgrade` (or `rm -rf yarn.lock && yarn`) and `resolutions` ([see here](https://yarnpkg.com/lang/en/docs/selective-version-resolutions/)).

Also, I wasn't trying to suggest that ember-source itself depended on `@ember/test-helpers`. I meant that there is an issue when using older `@ember/test-helpers` along with `ember-source@3.3` that we fixed in `@ember/test-helpers@0.7.22`.

**Comment by Frozenfire92** [comment]
oh ok, thanks for the clarification!

**Comment by dbbk** [comment]
Just an FYI I've been stuck on this problem for the last few days until seeing your fix to remove and re-add `ember-cli-qunit` @rwjblue. I just used `ember-cli-update` to upgrade from 3.1 and the test helpers stopped working properly.

I don't know if there's a simple solution but I can imagine a lot of others will have gotten stuck on this.
